--- TagLibraryInfoImpl.java 2013-10-01 00:00:00.000000000 +0200
+++ TagLibraryInfoImpl.java.modified    2013-10-01 00:00:00.000000000 +0200
@@ -20,10 +20,12 @@
 import java.io.File;
 import java.io.FileInputStream;
 import java.io.FileNotFoundException;
+import java.io.IOException;
 import java.io.InputStream;
 import java.io.PrintWriter;
 import java.io.StringWriter;
 import java.net.URL;
+import java.net.URLConnection;
 import java.util.Collection;
 import java.util.Enumeration;
 import java.util.Hashtable;
@@ -111,8 +113,19 @@
     // the following is a workaround until these problems are resolved.
     private InputStream getResourceAsStream(String uri)
             throws FileNotFoundException {
-        // Is uri absolute?
-        if (uri.startsWith("file:")) {
+
+        if (uri.startsWith("bundle")) {
+            URL url;
+            try {
+                url = new URL(uri);
+                URLConnection urlConnection = url.openConnection();
+                urlConnection.setUseCaches(false);
+                return urlConnection.getInputStream();
+            } catch (IOException e) {
+                throw new IllegalArgumentException("Unable to load URI '" + uri + "'", e);
+            }
+        } else if (uri.startsWith("file:")) {
+            // Is uri absolute?
             return new FileInputStream(new File(uri.substring(5)));
         } else {
             try {
